src/components/common/FilterGroup/index.tsx

```diff
@@ -0,0 +1,199 @@
+import FilterNumberRange from './filters/FilterNumberRange';
+import FilterCountWeight from './filters/FilterCountWeight';
+import { Option } from '@terraware/web-components/components/table/types';
+import { makeStyles } from '@mui/styles';
+import { Button } from '@terraware/web-components';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+
+interface StyleProps {
+  isMobile?: boolean;
+  isDesktop?: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  divider: {
+    maxWidth: '90%',
+    border: 'none',
+    borderTop: `1px solid ${theme.palette.TwClrBrdrTertiary}`,
+  },
+  button: {
+    width: (props: StyleProps) => (props.isMobile ? '100%' : 'auto'),
+  },
+}));
+
+export type FilterField = {
+  name: string;
+  label: string;
+  type:
+    | 'multiple_selection'
+    | 'single_selection'
+    | 'search'
+    | 'date_range'
+    | 'number_range'
+    | 'count_weight'
+    | 'hidden';
+};
+
+export type FilterGroupProps = {
+  initialFilters: Record<string, SearchNodePayload>;
+  fields: FilterField[];
+  values: FieldValuesPayload;
+  onConfirm: (filters: Record<string, SearchNodePayload>) => void;
+  onCancel: () => void;
+};
+
+export default function FilterGroup(props: FilterGroupProps): JSX.Element {
+  const { initialFilters, fields, values, onConfirm, onCancel } = props;
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+  const classes = useStyles({ isMobile });
+
+  // the filters defined by this filter group
+  const [filters, setFilters] = useState<Record<string, SearchNodePayload>>(initialFilters);
+
+  const onFilterChange = (key: string, filter: SearchNodePayload) => {
+    const newFilt: Record<string, SearchNodePayload> = {};
+    newFilt[key] = filter;
+    setFilters({ ...filters, ...newFilt });
+  };
+  const onDeleteFilter = (key: string) => {
+    const newFilters = { ...filters };
+    delete newFilters[key];
+    setFilters(newFilters);
+  };
+  const clearFilters = () => setFilters({});
+
+  return (
+    <Box display='flex' flexDirection='column' maxHeight='90vh'>
+      <Box
+        display='flex'
+        alignItems='center'
+        justifyContent='left'
+        width='100%'
+        borderBottom={`1px solid ${theme.palette.TwClrBrdrTertiary}`}
+        borderRadius={theme.spacing(1, 1, 0, 0)}
+        padding={theme.spacing(2, 3)}
+        sx={{
+          background: theme.palette.TwClrBgSecondary,
+        }}
+      >
+        <Typography fontSize='20px' fontWeight={600}>
+          {strings.FILTERS}
+        </Typography>
+      </Box>
+      <Box flex='1 1 auto' overflow='auto'>
+        {fields.map((f, index) => (
+          <Box key={f.name}>
+            {index > 0 && <hr className={classes.divider} />}
+            <Typography fontSize='14px' fontWeight={600} margin={theme.spacing(2, 2, 0, 2)}>
+              {f.label}
+            </Typography>
+            {f.type === 'multiple_selection' && (
+              <MultipleSelection
```

Comment:
A lot of these filter types will end up showing user defined values - which could be exhaustive. wondering if we need a scrollable sub filter area  - but that leads to a double scroll ux pattern (parent container scrolls). just calling that out since maybe the design didn't account for it

Reply:
I'm going to be replacing the multiple selection checkboxes with the FilterMultiSelect, which should clean things up for that. The single select may still be exhaustive and large, so i can replace it with a Dropdown select. Those two changes should take care of this.

Reply:
Awesome 🙌